spec/unit/outputs/elasticsearch/data_stream_support_spec.rb

```diff
@@ -152,6 +152,42 @@
 
   end
 
+  context 'ds value-dependent configuration' do
+    # Valid settings values
+    let(:options) { super().merge(
+      'action' => 'create',
+      'routing' => 'any',
+      'pipeline' => 'any',
+      'manage_template' => 'false',
+      'data_stream' => 'true')
+    }
+
+    context 'with valid values' do
+      let(:options) { super().merge(
+        'data_stream_type' => 'logs',
+        'data_stream_dataset' => 'any',
```

Comment:
Are all of these necessary for the functionality under test?

Reply:
Not really, testing only one would be enough as the `invalid_data_stream_params` method is only checking if the name prefix starts with `data_stream_*`. I ended up adding all valid `data_stream_` options here as a safeguard. Do you think we should keep only one?

Reply:
So maybe the name of the `context` could be `when any data_stream option has a valid value` instead of `with valid values`

Reply:
It's testing all possible valid values, not only the `data_stream` ones. It's merging the `data_stream_*` values with the parent context `let(:options) { 'action' => ..}` values.
Giving it a second thought, it's actually confusing, maybe it would be better to remove this parent `let(:options)` and move everything into the `with valid values` context. WDYT?